### ISSUE TITLE
Incorrect BCP47 language tag matching?

### ISSUE THREAD

**Issue by danielweck** [readium/readium-desktop#206]
https://github.com/readium/readium-desktop/blob/21cb8fd30c8dbae9c570fe0cd8d2e1c952795dff/src/common/services/translator.ts#L62-L82

```javascript
if (field[this.locale]) {
  return field[this.locale];
}
```

This current code does not match BCP47 language tags correctly, for example it will miss "en-uk" vs. "en-us" if `this.locale` is set to "en".

**Comment by danielweck** [comment]
Example with EPUB "kusamakura japanese vertical writing":

https://github.com/IDPF/epub3-samples/tree/master/30/kusamakura-japanese-vertical-writing
( https://github.com/IDPF/epub3-samples/releases/download/20170606/kusamakura-japanese-vertical-writing.epub )

See the usage of "language map" objects for `ja-xxx` locales (in particular, `author` and `title`, but also see `narrator` and `contributor` for examples of mixed content types):

https://readium2.herokuapp.com/pub/aHR0cHM6Ly9yYXcuZ2l0aHVidXNlcmNvbnRlbnQuY29tL0lEUEYvZXB1YjMtc2FtcGxlcy9tYXN0ZXIvMzAva3VzYW1ha3VyYS1qYXBhbmVzZS12ZXJ0aWNhbC13cml0aW5nL01FVEEtSU5GL2NvbnRhaW5lci54bWw%3D/manifest.json/show/all

Alternative link (in case the above fails):

https://readium2.now.sh/pub/aHR0cHM6Ly9yYXcuZ2l0aHVidXNlcmNvbnRlbnQuY29tL0lEUEYvZXB1YjMtc2FtcGxlcy9tYXN0ZXIvMzAva3VzYW1ha3VyYS1qYXBhbmVzZS12ZXJ0aWNhbC13cml0aW5nL01FVEEtSU5GL2NvbnRhaW5lci54bWw%3D/manifest.json/show/all

**Comment by danielweck** [comment]
Also note that in addition to `translateContentField()` (`src/common/services/translator.ts`) there is also `convertMultiLangStringToString()` which is a legacy / outdated implementation (see `FIXME`):

https://github.com/readium/readium-desktop/blob/21cb8fd30c8dbae9c570fe0cd8d2e1c952795dff/src/common/utils.ts#L22-L37

```javascript
    // FIXME: returns the string for a given languae
    const lang = langs[0];
    return item[lang];
```

**Comment by danielweck** [comment]
Related issue: https://github.com/readium/readium-desktop/issues/205

**Comment by llemeurfr** [comment]
Manuel, please check that this issue is now solved.

**Comment by Scarsniik** [comment]
fixed in dc6e138
